src/utils/resolve_url.ts

```diff
@@ -122,4 +68,192 @@ function getFilenameIndexInUrl(url: string): number {
   return indexOfLastSlash + 1;
 }
 
+/**
+ * Resolve the output URL from the baseURL and the relative reference as
+ * specified by RFC 3986 section 5.
+ * @param base
+ * @param relative
+ * @see https://datatracker.ietf.org/doc/html/rfc3986#section-5
+ * @example base: http://example.com | relative: /b/c | output: http://example.com/b/c
+ * @returns the resolved url
+ */
+function _resolveURL(base: string, relative: string) {
+  const baseParts = parseURL(base);
+  const relativeParts = parseURL(relative);
+
+  if (relativeParts.scheme) {
+    return formatURL(relativeParts);
+  }
+
+  const target: ParsedURL = {
+    scheme: baseParts.scheme,
+    authority: baseParts.authority,
+    path: "",
+    query: relativeParts.query,
+    fragment: relativeParts.fragment,
+  };
+
+  if (relativeParts.authority) {
+    target.authority = relativeParts.authority;
+    target.path = removeDotSegment(relativeParts.path);
+    return formatURL(target);
+  }
+
+  if (relativeParts.path === "") {
+    target.path = baseParts.path;
+    if (!relativeParts.query) {
+      target.query = baseParts.query;
+    }
+  } else {
+    if (startsWith(relativeParts.path, "/")) {
+      // path is absolute
+      target.path = removeDotSegment(relativeParts.path);
+    } else {
+      // path is relative
+      target.path = removeDotSegment(mergePaths(baseParts, relativeParts.path));
+    }
+  }
+  return formatURL(target);
+}
+
+interface ParsedURL {
```

Comment:
We generally prefix every TypeScript type without an influence on Javascript (so `type`s and `interface`s basically, but not enums / const enums) with the letter `I`